assembl/static2/js/app/components/login/legalForm.jsx

```diff
@@ -0,0 +1,86 @@
+// @noflow
+import * as React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Modal, Button } from 'react-bootstrap';
+import { closeModal } from '../../utils/utilityManager';
+
+type LegalFormProps = {
+  checked: boolean,
+  handleAcceptButton: () => void,
+  style?: Object
+};
+
+type LegalFormState = {
+  isScrolled: boolean
+};
+
+class LegalForm extends React.Component<LegalFormProps, LegalFormState> {
+  box: ?HTMLElement;
+
+  static defaultProps = {
+    style: {}
+  };
+
+  state = {
+    isScrolled: false
+  };
+
+  componentDidMount() {
+    this.box.addEventListener('scroll', this.trackScrolling);
+  }
+
+  trackScrolling = () => {
+    const wrappedElement = this.box;
+    if (wrappedElement.scrollHeight - wrappedElement.scrollTop === wrappedElement.clientHeight) {
+      this.setState({
+        isScrolled: true
+      });
+      wrappedElement.removeEventListener('scroll', this.trackScrolling);
+    }
+  };
+
+  handleSubmit = (legalContentsType: string) => {
+    this.props.handleAcceptButton(legalContentsType);
+    closeModal();
+  };
+
+  render() {
+    const { isScrolled } = this.state;
+    const { checked, text, style, legalContentsType } = this.props;
+    const boxClasses = checked ? 'terms-box justify full-height' : 'terms-box justify';
+
+    return (
+      <div className="terms-form" style={style}>
+        <Modal.Header closeButton>
+          <Modal.Title>
+            <Translate value={`${legalContentsType}.headerTitle`} />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div
+            className={boxClasses}
+            ref={(box) => {
+              this.box = box;
```

Comment:
note that you can now use the "new" `React.createRef` API (that's just a remark, you don't need to change it here)